### PR TITLE
Some comments to chttp2 transport 

### PR DESCRIPTION
**Pull request by sreecha**
Add some comments to chttp2_transport.cc file to explain the write path better

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]      +8  +0.0%

  [ = ]       0 TOTAL       +8  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,950,932      Total (=)      1,950,932

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,662,138      Total (<)     10,662,144

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc [hunk 1]
```diff
@@ -1014,6 +1036,7 @@ static void write_action(void* gt, grpc_error* error) {
                         grpc_combiner_scheduler(t->combiner)));
 }
 
+/* Callback from the grpc_endpoint after bytes have been written on the wire */
```
Comment (yashykt) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.cc]: saying 'on the wire' is wrong

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc [hunk 2]
```diff
@@ -812,6 +812,12 @@ static void set_write_state(grpc_chttp2_transport* t,
                                  write_state_name(t->write_state),
                                  write_state_name(st), reason));
   t->write_state = st;
+  // If the state is being reset back to idle, it means a write was just
```
Comment (yashykt) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.cc]: Pick one commenting strategy.. // or /*.. */

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc (hunk 1 above)
Comment (sreecha) [reply src/core/ext/transport/chttp2/transport/chttp2_transport.cc]: what should this be instead? - should we say after calling `sendmsg` ?

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc (hunk 2 above)
Comment (sreecha) [reply src/core/ext/transport/chttp2/transport/chttp2_transport.cc]: good point.Done

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]      +8  +0.0%

  [ = ]       0 TOTAL       +8  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by sreecha** [comment]
You are right. Fixed the comments. At least it should be consistent within a given PR. :)

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]      +8  +0.0%

  [ = ]       0 TOTAL       +8  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,950,932      Total (=)      1,950,932

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,662,144      Total (>)     10,662,139

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,950,932      Total (=)      1,950,932

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,662,142      Total (>)     10,662,136

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by sreecha** [comment]
The Mac failures in `grpclb_end2end_test` is known to be flaky (on MacOS). #15027

**Pull request merged**